web/src/app/alerts/CreateAlertDialog/CreateAlertDialog.js

```diff
@@ -0,0 +1,150 @@
+import React, { useState, useEffect } from 'react'
+import {
+  makeStyles,
+  Button,
+  Grid,
+  Stepper,
+  Step,
+  StepLabel,
+  Typography,
+  Link,
+} from '@material-ui/core'
+import OpenInNewIcon from '@material-ui/icons/OpenInNew'
+
+import { useCreateAlerts } from './useCreateAlerts'
+import { fieldErrors, allErrors } from '../../util/errutil'
+import FormDialog from '../../dialogs/FormDialog'
+import { CreateAlertForm } from './StepContent/CreateAlertForm'
+import { CreateAlertReview } from './StepContent/CreateAlertReview'
+
+import _ from 'lodash-es'
+
+const stepTitles = ['Alert Info', 'Service Selection', 'Confirm']
+const pluralize = num => (num !== 1 ? 's' : '')
+
+const useStyles = makeStyles(theme => ({
+  dialog: {
+    [theme.breakpoints.up('md')]: {
+      height: '65vh',
+    },
+  },
+  flexGrow: {
+    flexGrow: 1,
+  },
+}))
+
+export default function CreateAlertDialog(props) {
+  const classes = useStyles()
+  const [step, setStep] = useState(0)
+  const [value, setValue] = useState({
+    summary: '',
+    details: '',
+    serviceIDs: props.serviceID ? [props.serviceID] : [],
+  })
+  const [mutate, { data, loading, error }, getSvcID] = useCreateAlerts(value)
+
+  const fieldErrs = fieldErrors(error)
+  const hasValidationError = fieldErrs.some(e =>
+    ['summary', 'details'].includes(e.field),
+  )
+
+  useEffect(() => {
+    if (hasValidationError) {
+      setStep(0)
+    }
+  }, [hasValidationError])
+
+  const hasSubmitted = Boolean(data) && !hasValidationError
+  const currentStep = loading ? 2 : step
+
+  let review, reviewTitle
+  if (hasSubmitted) {
+    const createdAlertIDs = _.chain(data)
+      .values()
+      .filter()
+      .map(a => a.id)
+      .value()
+
+    const failedServices = allErrors(error).map(e => ({
+      id: getSvcID(e.path),
+      message: e.message,
+    }))
+
+    const failMessage = failedServices.length
+      ? `(${failedServices.length} failed)`
+      : ''
+
+    const titleMessage = `Successfully created ${
+      createdAlertIDs.length
+    } alert${pluralize(createdAlertIDs.length)} ${failMessage}`
+
+    reviewTitle = (
+      <Grid container>
+        <Grid item>
+          <Typography>{titleMessage}</Typography>
+        </Grid>
+        <Grid item className={classes.flexGrow} />
+        <Grid item>
+          <Link
+            href={`/alerts?allServices=1&filter=all&search=${encodeURIComponent(
+              value.summary,
+            )}`}
+            target='_blank'
+            rel='noopener noreferrer'
+          >
```

Comment:
Is it necessary to use the Link component here? The Button component has an href prop
https://material-ui.com/api/button/